proposal.rst

```diff
@@ -327,16 +335,18 @@ FILTER :
 
 	then we just return probability of L times the probability of R times the total range, which would be the union of the two ranges.
 
-NEAR, and PHRASE :
+NEAR(nearpostlist), PHRASE(phrasepostlist), and exactphrasepostlist :
 
-	No work needed on this.
+	implementing get_used_docid_range().
+
+	I think we can treat it like MULTI_AND, and the divide that estimate by constant like what we doing now.
```

Comment:
The way these 3 work is the same.  Considering NearPostList, this just performs the positional check, taking a stream of postings to check from a sub-postlist object.  The subqueries of the OP_NEAR are combined with other adjacent "and-like" queries, and the positional check is hoisted as high as we can, so that it has to check fewer documents.

So e.g. `(a AND b) AND (c NEAR d)` -> NearPostList(MultiAndPostList(a, b, c, d))` (and the NearPostList knows that it should check the positions of `c` and `d`.

I think this means that most (maybe all) of the work of using the ranges is done for you by the `MultiAndPostList` inside the `NearPostList`.